pkg/types/chain_writer.go

```diff
@@ -10,9 +10,9 @@ import (
 type ChainWriter interface {
 	// SubmitTransaction packs and broadcasts a transaction to the underlying chain.
 	//
-	// - `args` specifies input parameters to the contract call.
+	// - `args` should be any object which maps a set of method param into the contract and method specific method params.
```

Comment:
Did we lose some work here? ChainReader has a much more comprehensive comment, and ChainWriter should either have the same, or we should be using a shared `type SomeAlias any` with the doc in one place.

Reply:
Yeah, the `args` param changed and the comment didn't get updated accordingly. Probably my fault.

Reply:
I'll update the docs in a follow up PR.